test/unit/node/update.test.ts

```diff
@@ -23,6 +26,56 @@ describe("update", () => {
       return response.end(JSON.stringify(latest))
     }
 
+    if (request.url === "/reject-status-code") {
+      response.writeHead(500)
+      return response.end("rejected status code test")
+    }
+
+    if (request.url === "/no-location-header") {
+      response.writeHead(301)
+      response.writeHead(301, "testing", {
+        location: "",
+      })
+      return response.end("rejected status code test")
+    }
+
+    if (request.url === "/location-redirect") {
+      const latest: LatestResponse = {
+        name: "4.0.2",
+      }
+      response.writeHead(200)
+      return response.end(JSON.stringify(latest))
+    }
+
+    if (request.url === "/with-location-header") {
+      response.writeHead(301)
+      response.writeHead(301, "testing", {
+        location: "/location-redirect",
+      })
+
+      return response.end()
+    }
+
+    // NOTES subpath match /redirect/${number}
+    // if number === 0, resolve it
+    if (request.url.match(/\/redirect\/[0-9]/)) {
```

Comment:
Or maybe `[0-9/]+$` so the trailing slash is optional.  But this is definitely overkill anyway since it is just for our tests so I would ignore this if I was you :stuck_out_tongue:

Reply:
You could also use regex matching to pull out the number rather than splitting but again non-blocking, just a thought.

Reply:
LOL I always have to look up RegEx expressions. I'll go with `[0-9]+/?$`.

> You could also use capture groups

I'm less familiar with those but I'll look into it! Sounds kind nice for readability.

Reply:
To get you kickstarted (feel free to merge as-is though):

```
const matches = request.url.match(/\/redirect\/([0-9]+)\/?$/))
if (matches) {
  const count = parseInt(matches[1])
}
```

This is untested haha but it should be something like this.